.github/workflows/check-publish.yml

```diff
@@ -0,0 +1,42 @@
+name: check-publish
+# This workflow checks that the libraries can be published on crates.io.
+permissions:
+  contents: read
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+env:
+  CARGO_TERM_COLOR: always
+jobs:
+  check-publish:
+    name: Check publish on crates.io
```

Comment:
That looks cool!

Reply:
But I would change the pr's name to `ref: ...`

Reply:
@qalisander how about `ref: new workspace structure`